### PR TITLE
Adds type checking for api with ts lint

### PR DESCRIPTION
**Pull request by aeimer**
## Description:
Adds Type Checking in tslint for API

## Improvements
- `'no-use-before-declare' rule requires type information` now gets checked

## Known Issues:
_NONE_

------
_Remember to prefix your PR-Title with `⚠ WIP: ` if you still work on it.
If you have reached a final state, remove the prefix._

### CODE REVIEW

File: api/src/models/Course.ts [hunk 1]
```diff
@@ -131,6 +131,7 @@ courseSchema.statics.importJSON = async function (course: ICourse, admin: IUser,
   course.lectures = [];
 
   try {
```
Comment (khase) [inline api/src/models/Course.ts]: maybe add a Comment why u need to disable this check here

File: api/src/models/units/Unit.ts [hunk 2]
```diff
@@ -74,12 +74,14 @@ unitSchema.statics.importJSON = async function(unit: IUnit, courseId: string, le
   unit._course = courseId;
 
   try {
```
Comment (khase) [inline api/src/models/units/Unit.ts]: same as above

File: api/src/models/Lecture.ts [hunk 3]
```diff
@@ -78,6 +78,7 @@ lectureSchema.statics.importJSON = async function(lecture: ILecture, courseId: s
   lecture.units = [];
 
   try {
```
Comment (khase) [inline api/src/models/Lecture.ts]: same as above

File: api/src/models/Course.ts [hunk 4]
```diff
@@ -131,6 +131,7 @@ courseSchema.statics.importJSON = async function (course: ICourse, admin: IUser,
...
+    // tslint:disable:no-use-before-declare
```
Comment (khase) [inline api/src/models/Course.ts]: maybe add a Comment why u need to disable this check here

File: api/src/models/Lecture.ts [hunk 5]
```diff
@@ -78,6 +78,7 @@ lectureSchema.statics.importJSON = async function(lecture: ILecture, courseId: s
...
+    // tslint:disable:no-use-before-declare
```
Comment (khase) [inline api/src/models/Lecture.ts]: same as above

File: api/src/models/units/Unit.ts [hunk 6]
```diff
@@ -74,12 +74,14 @@ unitSchema.statics.importJSON = async function(unit: IUnit, courseId: string, le
...
+    // tslint:disable:no-use-before-declare
```
Comment (khase) [inline api/src/models/units/Unit.ts]: same as above

**Comment by aeimer** [comment]
@grosspersky @lukaskorte shall we merge?

**Pull request merged**